akamai/cache-client.go

```diff
@@ -283,7 +282,10 @@ func (cpc *CachePurgeClient) authedRequest(endpoint string, body v3PurgeRequest)
 	return nil
 }
 
-func (cpc *CachePurgeClient) purgeBatch(queueEntries [][]string) error {
+// Purge dispatches the provided queue entries in a request to the Akamai
+// Fast-Purge API. The request will be attempted cpc.retries number of times
+// before giving up and returning ErrAllRetriesFailed.
+func (cpc *CachePurgeClientImpl) Purge(queueEntries [][]string) error {
```

Comment:
Since this will purge (or fail to purge) all the queueEntries as a single unit, it would be a cleaner API for its parameter to be a single flattened slice of strings. The caller (which cares about certain clusters of URLs being purged together) can do the flattening.